base_layer/core/src/mempool/unconfirmed_pool/unconfirmed_pool.rs

```diff
@@ -116,7 +118,7 @@ impl UnconfirmedPool {
             }
             self.txs_by_priority
                 .insert(prioritized_tx.priority.clone(), tx_key.clone());
-            self.txs_by_signature.insert(tx_key, prioritized_tx);
+            self.txs_by_signature.insert(tx_key.clone(), prioritized_tx);
```

Comment:
Same amount of clones as before (unless `self.txs_by_signature.contains_key(tx_key) == true` then no clones) - tx_key is a reference.